public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,879 @@
+  a choice [revisited below](#form-array "Form arrays").
+
+  Nonetheless, the two models are pretty close in shape and you'll see in a moment how this alignment facilitates copying the _data model_ properties
+  to the _form model_ with the `patchValue` and `setValue` methods.
+
+:marked
+  Take a moment to refactor the _address_ `FormGroup` definition for brevity and clarity as follows:
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'address-form-group')(format=".")
+:marked
+
+.l-main-section
+a#set-data
+:marked
+  ## Populate the form model with _setValue_ and _patchValue_
+  You initialize control values in a `FormGroup` using `setValue` or `patchValue`.
+
+  ### _setValue_
+  With **`setValue`**, you assign _every_ form control value _at once_
+  by passing in a data object whose properties exactly match the _form model_ behind the `FormGroup`.
+
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'set-value','app/hero-detail.component.ts (excerpt)')(format=".")
+:marked
+  The `setValue` method checks the data object thoroughly before assigning any form control values.
+  It rejects a data object that doesn't match the `FormGroup` structure.
+  It rejects a data object if it is missing values for any control in the group.
+  It does return helpful errors.
+
+  Notice that you can _almost_ use the entire `hero` as the argument to `setValue`
+  because its shape is similar to the component's `FormGroup` structure.
+
+  You can only show the hero's first address and you must account for the possibility that the `hero` has no addresses at all.
+  This explains the conditional setting of the `address` property in the data object argument:
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'set-value-address')(format=".")
+
+:marked
+  ### _patchValue_
+  With **`patchValue`**, you can assign values to specific controls in a `FormGroup`
+  by supplying an object of key/value pairs for just the controls of interest. 
+
+  This example sets only the form's `name` control.
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'patch-value','app/hero-detail.component.ts (excerpt)')(format=".")
+
+:marked
+  With **`patchValue`** you have more flexibility to cope with wildly divergent data and form models.
+  But unlike `setValue`,  `patchValue` cannot check for missing control values and does not return errors.
+
+  ### When to set form model values (_ngOnChanges_)
+
+  Now you know _how_ to set the _form model_ values. But _when_ do you set them?
+  The answer depends upon when the component gets the _data model_ values.
+  
+  If the `HeroDetailComponent` is editing a _new_ hero, the data model consists entirely of default values
+  that you could copy to the `FormControls` when you create them in the component's constructor.
+  That's effectively what you did when you set each control to the empty string.
+
```

Comment:
Is it a common enough scenario to also mention the routing Resolve to prefetch the data? That was a common technique in Angular 1 with the UI Router.